src/edu/csus/ecs/pc2/ui/SubmitSampleRunsPane.java

```diff
@@ -0,0 +1,877 @@
+                if (selectedJudgingTypeIndexes == null || selectedJudgingTypeIndexes.length == 0) {
+                    FrameUtilities.showMessage(this, "No Judging Types selected", "No Judging Types selected, select at least one");
+                    return;
+                }
+
+                // fitler by judging type (accepted, wrong_answer, etc.)
+                files = ListUtilities.filterByJudgingTypes(files, submissionSolutionList);
+            }
+        }
+	    
+        if (checkBoxProblems.isSelected() ) {
+            if (submitSelectedProblemsRadioButton.isSelected()) {
+                // submissions may be filtered by problem
+
+                if (selectedProblemsIndexes == null || selectedProblemsIndexes.length == 0) {
+                    FrameUtilities.showMessage(this, "No Problem selected", "No Problems selected, select at least one");
+                    return;
+                }
+
+                // filter list by problem name
+                files = ListUtilities.filterByProblems (files, selectedProblemList);
+
+            }
+        }
+	    
+	    if (checkBoxLanguage.isSelected()) {
+	        // submissions may be filtered by language
+	        
+	        if (submitSelectedLanguagesRadioButton.isSelected()) {
+	            if (selectedLanguagesIndexes == null || selectedLanguagesIndexes.length == 0) {
+	                FrameUtilities.showMessage(this, "No Language selected", "No Languages selected, select at least one");
+	                return;
+
+	            }
+
+	            files = ListUtilities.filterByLanguages(files, getContest(), selectedLanguageList);
+	        }
+	    }
+	    
+
+	    int count = 0;
+	    for (File file : files) {
+	        count++;
+	        xlog("Will submit #" + count + " file =" + file.getAbsolutePath());
+	    }
+
+	    if (count == 0) {
+	        showMessage("There are no CDP samples source files  under " + cdpTextField.getText());
+	        return;
+	    }
+
+	    int result = FrameUtilities.yesNoCancelDialog(this, "Submit " + files.size() + " sample submissions?",
+	            "Submit CDP submissions");
+
+	    if (result == JOptionPane.YES_OPTION) {
+	        /**
+	         * This will submit each file found
+	         */
+	        submitter.sendSubmissions(files);
+
+	        showMessage("Submitted " + files.size() + " runs.");
+	    }
+
+
+	}
+
+	/**
+	 * Check whether cdpDir and model match
+	 * @param cdpDir base dir for CDP, parent dir for config/ dir
+	 * @return null if no issues, else a warning message about a diffence between model and cdpDir
+	 */
+	private String verifyCDP(String cdpDir) {
+		
+		// TODO compare problems in CDP with problems in model 
+		
+		return null;
+	}
+
+	/**
+	 * Returns the list of filenames that end in extension
+	 * 
+	 * @param files
+	 * @param extension
+	 * @return
+	 */
+	public static List<File> filterSource(List<File> files, String extension) {
```

Comment:
`filterSource()` does not seem like a particularly good descriptive name for what this method does.  The method applies a _very specific_ kind of filter (it filters out files whose names do not end with the specified extension).  This is accurately described in the JavaDoc comment in the method header; I think it should be equally described in the method name -- changing it to (something like) `filterByFilenameExtension`.